engine/schema/src/main/java/com/cloud/upgrade/SystemVmTemplateRegistration.java

```diff
@@ -0,0 +1,769 @@
+            pstmt.setString(1, hypervisorAndTemplateName.second());
+            try (ResultSet rs = pstmt.executeQuery()) {
+                if (rs.next()) {
+                    templateId = rs.getLong(1);
+                }
+            } catch (final SQLException e) {
+                String errMsg = String.format("updateSystemVmTemplates: Exception caught while getting ids of SystemVM templates: %s ", e.getMessage());
+                LOGGER.error(errMsg);
+                throw new CloudRuntimeException(errMsg);
+            }
+        } catch (SQLException e) {
+            String errorMessage = "Unable to upgrade the database";
+            LOGGER.error(errorMessage, e);
+            throw new CloudRuntimeException(errorMessage, e);
+        }
+        return templateId;
+    }
+
+    private static String fetchTemplatesPath() {
+            String filePath = relativeTemplatePath + metadataFileName;
+            LOGGER.debug("Looking for file [" + filePath + "] in the classpath.");
+            File metaFile = new File(filePath);
+            String templatePath = null;
+            if (metaFile.exists()) {
+                templatePath = relativeTemplatePath;
+            }
+            if (templatePath == null) {
+                filePath = AbsolutetemplatesPath + metadataFileName;
+                metaFile = new File(filePath);
+                templatePath = AbsolutetemplatesPath;
+                LOGGER.debug("Looking for file [" + filePath + "] in the classpath.");
+                if (!metaFile.exists()) {
+                    String errMsg = String.format("Unable to locate metadata file in your setup at %s", filePath.toString());
+                    LOGGER.error(errMsg);
+                    throw new CloudRuntimeException(errMsg);
+                }
+            }
+        return templatePath;
+    }
+
+    private static String getHypervisorName(String name) {
+        if (name.equals("xenserver")) {
+            return "xen";
+        }
+        if (name.equals("ovm3")) {
+            return "ovm";
+        }
+        return name;
+
+    }
+
+    private static Hypervisor.HypervisorType getHypervisorType(String hypervisor) {
+        if (hypervisor.equalsIgnoreCase("xen")) {
+            hypervisor = "xenserver";
+        } else if (hypervisor.equalsIgnoreCase("ovm")) {
+            hypervisor = "ovm3";
+        }
+        return Hypervisor.HypervisorType.getType(hypervisor);
+    }
+
+    private static List<Long> getEligibleZoneIds(Connection conn) {
+        List<Long> zones = new ArrayList<Long>();
+        try {
+            PreparedStatement pstmt = conn.prepareStatement(FETCH_DISTINCT_ELIGIBLE_ZONES);
+            ResultSet rs = pstmt.executeQuery();
+            while (rs.next()) {
+                zones.add(rs.getLong(1));
+            }
+        } catch (SQLException e) {
+            String errMsg = "Failed to fetch eligible zones for SystemVM template registration due to: %s";
```

Comment:
We could use `String.format` here, and then only passing the var to the log and the exception.